packages/ripple-binary-codec/tools/generateDefinitions.js

```diff
@@ -0,0 +1,360 @@
+if (process.argv.length != 3 && process.argv.length != 4) {
+  console.error(
+    'Usage: ' +
+      process.argv[0] +
+      ' ' +
+      process.argv[1] +
+      ' path/to/rippled [path/to/pipe/to]',
+  )
+  process.exit(1)
+}
+
+////////////////////////////////////////////////////////////////////////
+//  Get all necessary files from rippled
+////////////////////////////////////////////////////////////////////////
+const path = require('path')
+
+const fs = require('fs')
+function readFile(filename) {
+  try {
+    return fs.readFileSync(filename).toString('utf-8')
+  } catch (err) {
+    console.error(`Error reading file ${filename}:`, err.message)
+    process.exit(1)
+  }
+}
+
+const sfieldHeaderFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/SField.h'),
+)
+const sfieldMacroFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/detail/sfields.macro'),
+)
+const ledgerFormatsMacroFile = readFile(
+  path.join(
+    process.argv[2],
+    '/include/xrpl/protocol/detail/ledger_entries.macro',
+  ),
+)
+const terFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/TER.h'),
+)
+const transactionsMacroFile = readFile(
+  path.join(
+    process.argv[2],
+    '/include/xrpl/protocol/detail/transactions.macro',
+  ),
+)
+
+const capitalizationExceptions = {
+  NFTOKEN: 'NFToken',
+  URITOKEN: 'URIToken',
+  URI: 'URI',
+  UNL: 'UNL',
+  XCHAIN: 'XChain',
+  DID: 'DID',
+  ID: 'ID',
+  AMM: 'AMM',
+}
+
+// Translate from rippled string format to what the binary codecs expect
+function translate(inp) {
+  try {
+    if (inp.match(/^UINT/m))
```

Comment:
Isn't this script more appropriate to be placed inside rippled repository? It is very sensitive to changes in the rippled internal types.

Reply:
One example where they differ is `TxnSignature`.

Reply:
rippled doesn't need this script at all, though. This script is only used by tooling.

Reply:
that's true, yes. But this script/definitions.json file is essential for any serialization/deserialization/signing of the transactions. 

A developer updating the rippled types is more equipped to update this script rather than the dev-tooling team duplicating the effort. But it's not too big of a problem.

Reply:
While I agree that this script could go in rippled, it pairs really well with the model generation script(s), and those should live here.